File: templates/python/aoc_year_day.py
import sys
from common import timer, Puzzle


class Day<DAY>(Puzzle, year=<YEAR>, day=<DAY>):
    @staticmethod
    def clean_input(data):
        return data

    @timer(part=1)
    def solve_part_1(self):
        """Solution for part 1"""
        return None

    @timer(part=2)
    def solve_part_2(self):
        """Solution for part 2"""
        return None


if __name__ == "__main__":
    filename = "input.txt"
    if len(sys.argv) > 1:
        filename = sys.argv[1]
    Day<DAY>(filename=filename).solve_all()

File: templates/PHP/aoc_year_day.php
<?php
use \Common\get_data as get_data;

function solve_part_1($data) {
    return NULL;
}

function solve_part_2($data) {
    return NULL;
}

function main($filename) {
    $data = get_data($filename, <YEAR>, <DAY>);
    $part_1 = \Common\timer('solve_part_1', [$data], 1);
    $part_2 = \Common\timer('solve_part_2', [$data], 2);

    return [$part_1, $part_2];
}

if ($argc > 1) {
    $fn = $argv[1];
} else {
    $fn = "input.txt";
}

$result = \Common\timer('main', [$fn], 0);

?>
